cmd/geth/main.go

```diff
@@ -378,13 +378,25 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 	}
 	ethClient := ethclient.NewClient(rpcClient)
 
+	var ethService *eth.Ethereum
```

Comment:
Hey @muhammednagy this is unfortunately not valid. You can run a light-node without having the `eth.Ethereum` service started. @ligi correct me if I'm wrong but we mostly care about crashes in full client not in light clients, so I think its valid to move this code into the `if ctx.GlobalInt...` on line 383. 

To repro failure:
```
> make geth
> ./build/bin/geth --syncmode=light --goerli
...
Fatal: Failed to retrieve ethereum service: unknown service
```

Reply:
yea - it would mainly be interesting for full nodes - so I think we should go with what you proposed. But if there is a way to have it for both this would also be nice

Reply:
What about using a different database? is it possible?
We could probably store them in a file instead which will work for whatever type of node we are running.

Reply:
It's definitely possible, but I think we want to have it in the same database. Otherwise we just wind up with a separate set of problems, where we need to care about consistency over many files, and keeping that file in sync with when e.g people do `removedb` or removes the db manually.

Reply:
okay i think in that case a file would work best for all regardless of the type of node.
Should i go ahead with having a file?
If not then do you know which db i can use that would work for all kinds of nodes?

Reply:
Just use the leveldb instance we already have throughout the code for everything.